HPXMLtoOpenStudio/resources/utility_bills.rb

```diff
@@ -105,47 +47,39 @@ def self.marginal_rate_to_average_rate(marginal_rate, fixed_charge, household_co
     return marginal_rate + 12.0 * fixed_charge / household_consumption
   end
 
-  def self.get_state_code_to_padd
-    # https://www.eia.gov/tools/glossary/index.php?id=petroleum%20administration%20for%20defense%20district
-    padd_to_state_codes = { 'PADD 1A' => ['CT', 'MA', 'ME', 'NH', 'RI', 'VT'],
-                            'PADD 1B' => ['DE', 'DC', 'MD', 'NJ', 'NY', 'PA'],
-                            'PADD 1C' => ['FL', 'GA', 'NC', 'SC', 'WV', 'VA'],
-                            'PADD 2' => ['IA', 'IL', 'IN', 'KS', 'KY', 'MI', 'MN', 'MO', 'ND', 'NE', 'OH', 'OK', 'SD', 'TN', 'WI'],
-                            'PADD 3' => ['AL', 'AR', 'LA', 'MS', 'NM', 'TX'],
-                            'PADD 4' => ['CO', 'ID', 'MT', 'UT', 'WY'],
-                            'PADD 5' => ['AK', 'AZ', 'CA', 'HI', 'NV', 'OR', 'WA'] }
-
-    state_code_to_padd = {}
-    padd_to_state_codes.each do |padd, state_codes|
-      state_codes.each do |state_code|
-        state_code_to_padd[state_code] = padd
+  def self.get_eia_seds_rate(state_code, fuel_type)
+    msn_code_map = {
+      HPXML::FuelTypeElectricity => 'ESRCD',
+      HPXML::FuelTypeNaturalGas => 'NGRCD',
+      HPXML::FuelTypeOil => 'DFRCD',
+      HPXML::FuelTypePropane => 'PQRCD',
+      HPXML::FuelTypeCoal => 'CLRCD',
+      HPXML::FuelTypeWoodCord => 'WDRCD',
+      HPXML::FuelTypeWoodPellets => 'WDRCD'
+    }
+    unit_conv = {
+      HPXML::FuelTypeElectricity => 'kwh', # convert $/MBtu to $/kWh
+      HPXML::FuelTypeNaturalGas => 'therm', # convert $/MBtu to $/therm
+      HPXML::FuelTypeOil => 'gal_fuel_oil', # convert $/MBtu to $/gallons of oil
+      HPXML::FuelTypePropane => 'gal_propane', # convert $/MBtu to $/gallons of propane
+      HPXML::FuelTypeCoal => 'mbtu', # $/MBtu
+      HPXML::FuelTypeWoodCord => 'mbtu', # $/MBtu
+      HPXML::FuelTypeWoodPellets => 'mbtu' # $/MBtu
+    }
+
+    CSV.foreach(File.join(File.dirname(__FILE__), '../../ReportUtilityBills/resources/simple_rates/pr_all_update.csv'), headers: true) do |row|
+      next if row['State'].upcase != state_code.upcase # State
+      next if row['MSN'].upcase != msn_code_map[fuel_type] # EIA SEDS MSN code
+
+      if fuel_type == HPXML::FuelTypeCoal
+        seds_rate = row['2007'] # Use 2007 prices for coal. For 2008 forward, EIA assumes there is zero residential sector coal consumption in the United States, and SEDS does not estimate a price.
+      else
+        seds_rate = row.to_h.values.reverse.find { |rate| !rate.nil? && !rate.empty? } # If the rate for the latest year is unavailable, find the last non-nil rate.
       end
```

Comment:
Can we not simply collapse these into generic logic? Clearly a value of zero is not an actual fuel price. Untested suggestion:

```suggestion
      seds_rate = row.to_h.values.reverse.find { |rate| rate.to_f != 0 } # If the rate for the latest year is unavailable, find the last non-nil/non-zero rate.
```